x-pack/plugins/apm/server/lib/helpers/setup_request.ts

```diff
@@ -85,7 +74,14 @@ export async function setupRequest<TParams extends SetupRequestParams>(
     context.core.uiSettings.client.get(UI_SETTINGS.SEARCH_INCLUDE_FROZEN),
   ]);
 
-  const uiFiltersES = decodeUiFilters(query.uiFilters);
+  const uiFilters: UIFilters = {};
+  if (query.uiFilters) {
+    try {
+      Object.assign(uiFilters, JSON.parse(query.uiFilters));
```

Comment:
nit: What's the reasoning for using assign? Doesn't this work?

```suggestion
      uiFilters = JSON.parse(query.uiFilters)
```

If it's because of some typescript issue it might be better to extract it to a separate function again
```ts
function decodeUiFilters(uiFiltersEncoded?: string) {
  if (!uiFiltersEncoded) {
    return {};
  }
  return JSON.parse(uiFiltersEncoded);
}
```